homeassistant/components/octoprint/camera.py

```diff
@@ -1,17 +1,37 @@
 """Support for OctoPrint binary camera."""
 from __future__ import annotations
 
+from collections.abc import Callable
+from json import JSONDecodeError
+
+import httpx
 from pyoctoprintapi import OctoprintClient, WebcamSettings
 
+from homeassistant.components.camera import DEFAULT_CONTENT_TYPE, Camera, StreamType
+from homeassistant.components.generic.camera import (
+    CONF_CONTENT_TYPE,
+    CONF_FRAMERATE,
+    CONF_LIMIT_REFETCH_TO_URL_CHANGE,
+    CONF_STILL_IMAGE_URL,
+    CONF_STREAM_SOURCE,
+    GenericCamera,
```

Comment:
The GenericCamera isn't written or meant to be used as a helper for other integrations. If we want to allow that it needs to be refactored first to have a proper interface with keyword only arguments instead of passing a dict with config items. The class also needs to be exported from the generic package via `__all__` so it's clear that other integrations can import it.

We did the same changes for `MjpegCamera` earlier.

I'll check with the core team of we think it's good to be able to reuse the generic camera.

Reply:
Okay. We can take a look at what was done with `MjpegCamera` and try to make a PR to see what that looks like for `GenericCamera`.

Reply:
The conclusion from the core team is it's ok to use the generic camera integration but the camera class should be refactored or split so that it gets an interface that other integrations can use, per my comment above.